Tiltfile

```diff
@@ -96,6 +96,9 @@ def fixup_yaml_empty_arrays(yaml_str):
 
 def validate_auth():
     substitutions = settings.get("kustomize_substitutions", {})
+    for key in keys:
```

Comment:
I realize now this won't work because it's looking at the keys instead of the substitutions. Here's what this should be:

```
def validate_auth():
    substitutions = settings.get("kustomize_substitutions", {})
    os.environ.update(substitutions)
    for sub in substitutions:
        if sub[-4:] == "_B64":
            os.environ[sub[:-4]] = base64_decode(os.environ[sub])
    missing = [k for k in keys if not os.environ.get(k)]
    if missing:
        fail("missing kustomize_substitutions keys {} in tilt-setting.json".format(missing))
```

This will set all the variables in substitutions as environment variables and then use those to set the decoded equivalent variable if the substitution had a _B64 suffix.

Since we're doing `os.environ.update(substitutions)` early now, you can delete the line in `dev capz():` that does `os.environ.update(substitutions)` and the one before (lines 136-137).

Reply:
we also need to change line 193 to

```
    os.putenv('AZURE_CLIENT_SECRET_B64', base64_encode(os.environ.get("AZURE_CLIENT_SECRET")))
```

and remove the `substitutions = settings.get("kustomize_substitutions", {})` line above (line 192)

So it gets the secret directly from the environment variables

Reply:
Finally, let's add `echo_off=True` to all the local() functions that might print secrets so we don't output those in logs (I spotted 3 in the file, in `base64_encode`, `base64_decode` and `create_identity_secret`). For example:

```
    encode_blob = local("echo '{}' | tr -d '\n' | base64 - | tr -d '\n'".format(to_encode), quiet=True, echo_off=True)
```